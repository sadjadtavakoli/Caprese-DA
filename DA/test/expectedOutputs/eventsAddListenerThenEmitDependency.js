
#8 function adderFunction-test/unit_tests/eventsaddlistenerthenemitdependency.js-8-10 entered from eventsAddListenerThenEmitDependency.js-test/unit_tests/eventsaddlistenerthenemitdependency.js-1-15
#4 function  listner1-test/unit_tests/eventsaddlistenerthenemitdependency.js-4-6 entered throught event connection2 emitted by function eventsAddListenerThenEmitDependency.js-test/unit_tests/eventsaddlistenerthenemitdependency.js-1-15
#end Execution