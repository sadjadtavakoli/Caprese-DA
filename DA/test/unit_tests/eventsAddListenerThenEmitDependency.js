var events = require('events');
var eventEmitter = new events.EventEmitter();

function listner1() {
    //   do nothing 
}

function adderFunction() {
    eventEmitter.addListener('connection2', listner1);
}


adderFunction()

eventEmitter.emit("connection2")
