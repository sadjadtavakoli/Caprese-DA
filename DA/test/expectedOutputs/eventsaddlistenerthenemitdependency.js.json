{"adderFunction-test/unit_tests/eventsaddlistenerthenemitdependency.js-8-10":{"impacted":["t_1","t_8"]},"emitter-test/unit_tests/eventsaddlistenerthenemitdependency.js-12-14":{"impacted":["t_1","t_4","t_7"]},"keyMap":{"t_1":"eventsaddlistenerthenemitdependency.js-test/unit_tests/eventsaddlistenerthenemitdependency.js-1-17","t_4":"adderFunction-test/unit_tests/eventsaddlistenerthenemitdependency.js-8-10","t_7":"listner1-test/unit_tests/eventsaddlistenerthenemitdependency.js-4-6","t_8":"emitter-test/unit_tests/eventsaddlistenerthenemitdependency.js-12-14"}}